tests/e2e/e2e_test.go

```diff
@@ -243,12 +248,411 @@ var _ = Describe("odoe2e", func() {
 				runCmd("odo log nodejs")
 			})
 
+			It("should be able to create binary component", func() {
+				runCmd("curl -o " + tmpDir + "/sample-binary-testing-1.war " +
+					"https://gist.github.com/mik-dass/f95bd818ddba508ff76a386f8d984909/raw/e5bc575ac8b14ba2b23d66b5cb4873657e1a1489/sample.war")
+				runCmd("odo create wildfly --binary " + tmpDir + "/sample-binary-testing-1.war")
+				cmpList := runCmd("odo list")
+				Expect(cmpList).To(ContainSubstring("wildfly"))
+			})
+		})
+	})
+
+	Describe("updating the component", func() {
+		Context("When update is done", func() {
+			It("should update component from binary to binary", func() {
+				runCmd("curl -o " + tmpDir + "/sample-binary-testing-2.war " +
+					"'https://gist.github.com/mik-dass/f95bd818ddba508ff76a386f8d984909/raw/85354d9ee8583a9c1e64a331425eede235b07a9e/sample%2520(1).war'")
+				runCmd("odo update wildfly --binary " + tmpDir + "/sample-binary-testing-2.war")
+
+				// checking bc for updates
+				getBc := runCmd("oc get bc wildfly-" + appTestName + " -o go-template={{.spec.source.git.uri}}")
+				Expect(getBc).To(Equal(bootStrapSupervisorURI))
+
+				// checking for init containers
+				getDc := runCmd("oc get dc wildfly-" + appTestName + " -o go-template='" +
+					"{{range .spec.template.spec.initContainers}}" +
+					"{{.name}}{{end}}'")
+				Expect(getDc).To(ContainSubstring(initContainerName))
+
+				// checking for volumes
+				getDc = runCmd("oc get dc wildfly-" + appTestName + " -o go-template='" +
+					"{{range .spec.template.spec.volumes}}" +
+					"{{.name}}{{end}}'")
+				Expect(getDc).To(ContainSubstring("wildfly" + appRootVolumeName))
+
+				// checking for volumes mounts
+				getDc = runCmd("oc get dc wildfly-" + appTestName + " -o go-template='" +
+					"{{range .spec.template.spec.containers}}{{range .volumeMounts}}{{.name}}" +
+					"{{.name}}{{end}}{{end}}'")
+				Expect(getDc).To(ContainSubstring("wildfly" + appRootVolumeName))
+
+				// checking for source-type in dc
```

Comment:
You could create function rather than copy/pasting this to every test ;-)

Reply:
ok good idea :+1: